### PR TITLE
Add Spacing section to styleguide

### PR DESCRIPTION
**Pull request by rfriberg**
Added a section to the styleguide to detail our new spacing utility classes.  Preview: https://precog.mapzen.com/mapzen/styleguide/rfriberg/add-spacing-guidelines/design-elements.html#spacing

I'm wondering if I should add something about `headroom` and `footroom`, as well.  Thoughts?

### CODE REVIEW

**Comment by souperneon** [comment]
This is so great! 
I don't know the answer to that… it would be useful for people creating standalone webpages to be able to get some guidance on headroom/footroom and when to use the subclasses there, right?

File: src/site/design-elements/spacing/index.html [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    <li><code>2</code><span class="mx-1">-</span> 12px (<code>$spacer * .5</code>)</li>
+    <li><code>3</code><span class="mx-1">-</span> 24px (<code>$spacer</code>)</li>
+    <li><code>4</code><span class="mx-1">-</span> 36px (<code>$spacer * 1.5</code>)</li>
+    <li><code>5</code><span class="mx-1">-</span> 72px (<code>$spacer * 3</code>)</li>
+  </ul>
+
+  <p class="mt-4">
+    <strong>Examples of these classes</strong>
+  </p>
+  <pre>
```
Comment (louh) [inline src/site/design-elements/spacing/index.html:82]: I would recommend adding an example with a breakpoint, and example that sets all sides.

**Comment by louh** [comment]
Yeah I would say `headroom` and `footroom` (and guidelines about them) definitely belong here.

File: src/site/design-elements/spacing/index.html (hunk 1 above)
Comment (rfriberg) [reply src/site/design-elements/spacing/index.html:82]: 👍

**Comment by rfriberg** [comment]
Yep, I'm going to add a subsection on vertical spacing to talk about headroom and footroom.  I think it'll be a helpful reference for everyone.

**Comment by rfriberg** [comment]
Thanks for feedback, all!  I added a new vertical spacing section and more examples.  Check it out: https://precog.mapzen.com/mapzen/styleguide/rfriberg/add-spacing-guidelines/design-elements.html#spacing

**Review by souperneon — APPROVED** [review]

**Pull request merged**